packages/core/tracing/src/components/trace-viewer/TraceViewer.vue

```diff
@@ -0,0 +1,221 @@
+<template>
+  <Splitpanes
+    class="trace-viewer"
+    horizontal
+  >
+    <Pane
+      class="summary-pane"
+      min-size="20"
+    >
+      <div class="above-waterfall">
+        <WaterfallLegend />
+
+        <div class="url">
+          <div class="label">
+            {{ t('trace_viewer.url') }}
+          </div>
+          <div class="content">
+            <KCopy
+              copy-tooltip="Copy"
+              :text="url"
+            />
+          </div>
+        </div>
+      </div>
+
+      <div class="waterfall-wrapper">
+        <WaterfallView
+          :root-span="props.rootSpan"
+          @update:selected-span="handleUpdateSelectedSpan"
+        />
+      </div>
+    </Pane>
+
+    <Pane
+      class="detail-pane"
+      size="50"
+    >
+      <div v-if="selectedSpan && !spanNothingToDisplay">
+        <SpanEventList
+          v-if="selectedSpan.span.events && selectedSpan.span.events.length > 0"
+          :span="selectedSpan.span"
+        />
+        <SpanAttributeTable
+          v-if="selectedSpan.span.attributes.length > 0"
+          :span="selectedSpan.span"
+        />
+      </div>
+      <div
+        v-else
+        class="empty-state"
+      >
+        <template v-if="!selectedSpan">
```

Comment:
nit: I suggest reversing the if branch (`v-if="selectedSpan"`) for better readability and flow.

Reply:
Updated and shifted the branches